WordPress/src/main/java/org/wordpress/android/ui/history/HistoryDetailContainerFragment.java

```diff
@@ -100,85 +88,83 @@ public static HistoryDetailContainerFragment newInstance(final Revision revision
+            plugins.add(new CaptionShortcodePlugin(mBinding.visualContent));
+            plugins.add(new VideoShortcodePlugin());
+            plugins.add(new AudioShortcodePlugin());
+            plugins.add(new HiddenGutenbergPlugin(mBinding.visualContent));
+            mBinding.visualContent.setPlugins(plugins);
+
+            boolean isInVisualPreview = savedInstanceState != null
+                                        && savedInstanceState.getBoolean(KEY_IS_IN_VISUAL_PREVIEW);
+            mBinding.visualPreviewContainer.setVisibility(isInVisualPreview ? View.VISIBLE : View.GONE);
+            mBinding.diffPager.setVisibility(isInVisualPreview ? View.GONE : View.VISIBLE);
+            mBinding.previous.setVisibility(isInVisualPreview ? View.INVISIBLE : View.VISIBLE);
+            mBinding.next.setVisibility(isInVisualPreview ? View.INVISIBLE : View.VISIBLE);
+
+            refreshHistoryDetail(mBinding, adapter, mRevision);
+            resetOnPageChangeListener(mBinding, adapter);
+        }
 
-        mVisualTitle = rootView.findViewById(R.id.visual_title);
-        mVisualContent = rootView.findViewById(R.id.visual_content);
-
-        Drawable loadingImagePlaceholder = EditorMediaUtils.getAztecPlaceholderDrawableFromResID(
-                requireContext(),
-                org.wordpress.android.editor.R.drawable.ic_gridicons_image,
-                EditorMediaUtils.getMaximumThumbnailSizeForEditor(requireContext()));
-
-        mVisualContent.setImageGetter(new AztecImageLoader(requireContext(), mImageManager, loadingImagePlaceholder));
-        mVisualContent.setKeyListener(null);
-        mVisualContent.setTextIsSelectable(true);
-        mVisualContent.setCursorVisible(false);
-        mVisualContent.setMovementMethod(LinkMovementMethod.getInstance());
-
-        ArrayList<IAztecPlugin> plugins = new ArrayList<>();
-        plugins.add(new WordPressCommentsPlugin(mVisualContent));
-        plugins.add(new CaptionShortcodePlugin(mVisualContent));
-        plugins.add(new VideoShortcodePlugin());
-        plugins.add(new AudioShortcodePlugin());
-        plugins.add(new HiddenGutenbergPlugin(mVisualContent));
-        mVisualContent.setPlugins(plugins);
-
-        mVisualPreviewContainer = rootView.findViewById(R.id.visual_preview_container);
-
-        boolean isInVisualPreview = savedInstanceState != null
-                                    && savedInstanceState.getBoolean(KEY_IS_IN_VISUAL_PREVIEW);
-        mVisualPreviewContainer.setVisibility(isInVisualPreview ? View.VISIBLE : View.GONE);
-        mViewPager.setVisibility(isInVisualPreview ? View.GONE : View.VISIBLE);
-        mPreviousButton.setVisibility(isInVisualPreview ? View.INVISIBLE : View.VISIBLE);
-        mNextButton.setVisibility(isInVisualPreview ? View.INVISIBLE : View.VISIBLE);
-
-        refreshHistoryDetail();
-        resetOnPageChangeListener();
-
-        return rootView;
+        return mBinding.getRoot();
```

Comment:
💡 We're checking if `mBinding` is `null` on line `115` but we're using it here without checking it. I wonder if we should do something here if `mBinding` is `null` (e.g. throwing an exception) or if we can just drop the null check in line `115` 🤔

Reply:
Yeaaa, and thanks for this comment @RenanLukas , ideally I would like to drop the null check on line `115`. Unfortunately, when I do that, for some reason, the IDE is warning us about it, I just doesn't understand that `mBinding` is already set, just a few lines above: `Dereference of 'mBinding' may produce 'NullPointerException'`

Also, when I do that, that is, dropping the null check on line `115`, the IDE now starts warning us about line `163`, as expected, I don't know why it is warning us on that already, probably an IDE bug of some kind: `Method invocation 'getRoot' may produce 'NullPointerException' `

🤷 🤷 🤷

I ended-up choosing to throw an `IllegalStateException` and guard the `mBinding.getRoot()` call. Let me know what you think and again, thanks for this suggestion:

This is done here: 724f91b43d1342b4fc4a50da144359a202b9ca58